base/compiler/ssair/inlining.jl

```diff
@@ -1722,32 +1731,35 @@ function late_inline_special_case!(
     return nothing
 end
 
-function ssa_substitute!(idx::Int, @nospecialize(val), arg_replacements::Vector{Any},
+function ssa_substitute!(insert_node_before!,
+                         subst_inst::Instruction, @nospecialize(val), arg_replacements::Vector{Any},
                          @nospecialize(spsig), spvals::SimpleVector,
                          spvals_ssa::Union{Nothing, SSAValue},
-                         linetable_offset::Int32, boundscheck::Symbol, compact::IncrementalCompact)
-    compact.result[idx][:flag] &= ~IR_FLAG_INBOUNDS
-    compact.result[idx][:line] += linetable_offset
-    return ssa_substitute_op!(val, arg_replacements, spsig, spvals, spvals_ssa, boundscheck, compact, idx)
+                         linetable_offset::Int32, boundscheck::Symbol)
+    subst_inst[:flag] &= ~IR_FLAG_INBOUNDS
+    subst_inst[:line] += linetable_offset
+    return ssa_substitute_op!(insert_node_before!, subst_inst,
+        val, arg_replacements, spsig, spvals, spvals_ssa, boundscheck)
 end
 
-function insert_spval!(compact::IncrementalCompact, idx::Int, spvals_ssa::SSAValue, spidx::Int, do_isdefined::Bool)
-    ret = insert_node!(compact, SSAValue(idx),
+function insert_spval!(insert_node_before!, spvals_ssa::SSAValue, spidx::Int, do_isdefined::Bool)
+    ret = insert_node_before!(
         effect_free(NewInstruction(Expr(:call, Core._svec_ref, false, spvals_ssa, spidx), Any)))
     tcheck_not = nothing
     if do_isdefined
-        tcheck = insert_node!(compact, SSAValue(idx),
+        tcheck = insert_node_before!(
             effect_free(NewInstruction(Expr(:call, Core.isa, ret, Core.TypeVar), Bool)))
-        tcheck_not = insert_node!(compact, SSAValue(idx),
+        tcheck_not = insert_node_before!(
             effect_free(NewInstruction(Expr(:call, not_int, tcheck), Bool)))
     end
     return (ret, tcheck_not)
 end
 
-function ssa_substitute_op!(@nospecialize(val), arg_replacements::Vector{Any},
+function ssa_substitute_op!(insert_node_before!, subst_inst::Instruction,
```

Comment:
```suggestion
function ssa_substitute_op!(insert_node_before!::Inserter, subst_inst::Instruction,
```

Shouldn't this be annotated here and above in `ssa_substitute!`?